internal/pkg/cli/app_init.go

```diff
@@ -232,6 +235,15 @@ func (o *initAppOpts) validateAppName(name string) error {
 	if err != nil {
 		var noSuchAppErr *config.ErrNoSuchApplication
 		if errors.As(err, &noSuchAppErr) {
+			roleName := fmt.Sprintf("%s-adminrole", name)
+			tags, err := o.iamRoleManager.ListRoleTags(roleName)
+			// return nil when both adminrole and application does not exists then it is a valid app name
+			if err != nil {
+				return nil
+			}
+			if tags != nil {
+				return fmt.Errorf("application named %s already exists in another region", name)
```

Comment:
I think the error message should be different, depending whether there is an `copilot-application` tag. If there is such a tag, then the existing error message makes sense. Otherwise, it'd mean that there isn't an application - just the role - in which case the error message wouldn't make sense.

What do you think of the following? Does it overcomplicate it?

Example interface:
```go

type errAppAlreadyExistsInAccount struct
func (e *errAppAlreadyExistsInAccount) Error() string
func (e *errAppAlreadyExistsInAccount) RecommendedActions() string 
// I think our recommended actions here would be something like
// 1. If they just want to add a workspace, reusing that existing application, we will ask them to switch to that region, and run `copilot app init`.
// 2. If they think the existing application should be deleted, they need to switch to that region, and run `copilot app delete`.


type errStackSetAdminRoleExistsInAccount struct
func (e *errStackSetAdminRoleExistsInAccount) Error() string
func (e *errStackSetAdminRoleExistsInAccount) RecommendedActions() string 
// 1. We explain why that IAM role prevent application to create
// 2. We ask them to create the application with a different name, so that the IAM role name does not collide

// Here
if _, ok := tags[deploy.AppTag]; ok {
   return &errAppAlreadyExistsInAccount{}
}
return &errAppAlreadyExistsInAccount{}
```